plugins/obs-qsv11/obs-qsv265.c

```diff
@@ -0,0 +1,1118 @@
+OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+
+#include <stdio.h>
+#include <util/dstr.h>
+#include <util/darray.h>
+#include <util/platform.h>
+#include <obs-module.h>
+#include <obs-hevc.h>
+//#include <obs-startcode.h>
+#include <d3d11.h>
+#include <dxgi1_2.h>
+
+#ifndef _STDINT_H_INCLUDED
+#define _STDINT_H_INCLUDED
+#endif
+
+#include "QSV_HevcEncoder.h"
+#include <Windows.h>
+
+#define do_log(level, format, ...)                 \
+	blog(level, "[qsv encoder: '%s'] " format, \
+	     obs_encoder_get_name(obsqsv->encoder), ##__VA_ARGS__)
+
+#define warn(format, ...) do_log(LOG_WARNING, format, ##__VA_ARGS__)
+#define info(format, ...) do_log(LOG_INFO, format, ##__VA_ARGS__)
+#define debug(format, ...) do_log(LOG_DEBUG, format, ##__VA_ARGS__)
+
+static uint8_t *ff_find_startcode_internal(const uint8_t *p, const uint8_t *end)
+{
+	const uint8_t *a = p + 4 - ((intptr_t)p & 3);
+
+	for (end -= 3; p < a && p < end; p++) {
+		if (p[0] == 0 && p[1] == 0 && p[2] == 1)
+			return (uint8_t *)p;
+	}
+
+	for (end -= 3; p < end; p += 4) {
+		uint32_t x = *(const uint32_t *)p;
+
+		if ((x - 0x01010101) & (~x) & 0x80808080) {
+			if (p[1] == 0) {
+				if (p[0] == 0 && p[2] == 1)
+					return (uint8_t *)p;
+				if (p[2] == 0 && p[3] == 1)
+					return (uint8_t *)(p + 1);
+			}
+
+			if (p[3] == 0) {
+				if (p[2] == 0 && p[4] == 1)
+					return (uint8_t *)(p + 2);
+				if (p[4] == 0 && p[5] == 1)
+					return (uint8_t *)(p + 3);
+			}
+		}
+	}
+
+	for (end += 3; p < end; p++) {
+		if (p[0] == 0 && p[1] == 0 && p[2] == 1)
+			return (uint8_t *)p;
+	}
+
+	return (uint8_t *)(end + 3);
+}
+
+static uint8_t *obs_find_annexB_startcode(const uint8_t *p, const uint8_t *end)
+{
+	uint8_t *out = ff_find_startcode_internal(p, end);
+	if (p < out && out < end && !out[-1])
+		out--;
+	return out;
+}
+
+/* ------------------------------------------------------------------------- */
+
+struct obs_qsv {
+	obs_encoder_t *encoder;
+
+	qsv_param_t params;
+	qsv_t *context;
+
+	DARRAY(uint8_t) packet_data;
+
+	uint8_t *extra_data;
+	uint8_t *sei;
+
+	size_t extra_data_size;
+	size_t sei_size;
+
+	os_performance_token_t *performance_token;
+};
+
+/* ------------------------------------------------------------------------- */
+
+static CRITICAL_SECTION g_QsvCs;
+static unsigned short g_verMajor;
+static unsigned short g_verMinor;
+static int64_t g_pts2dtsShift;
+static int64_t g_prevDts;
+static bool g_bFirst;
+
+static const char *obs_qsv_hevc_getname(void *type_data)
+{
+	UNUSED_PARAMETER(type_data);
+	return "QuickSync HEVC Encoder";
```

Comment:
Remove "Encoder" for consistency with H.264.

Reply:
Done.